astropy/io/registry.py

```diff
@@ -414,10 +432,11 @@ def _is_best_match(class1, class2, format_classes):
       - OR class1 is a subclass of class2 and class1 is not in classes.
         In this case the subclass will use the parent reader/writer.
     """
-    classes = [cls for fmt, cls in format_classes]
-    is_best_match = ((class1 is class2) or (issubclass(class1, class2) and
-                                            class1 not in classes))
-    return is_best_match
+    # The set with the classes is only created if class1 is not class2 and
+    # class1 is a subclass of class2.
+    return (class1 is class2 or
+            (issubclass(class1, class2) and
+             class1 not in {cls for fmt, cls in format_classes}))
```

Comment:
I thought straight iterator with `()` might be faster yet, but a quick test on my python3 shows that is not true; this does seem the fastest...